File: generated_code/cpp/solution_2532_2.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool equalFrequency(string word) {
        // Count frequencies of each character
        vector<int> char_counts(26, 0);
        for (char c : word) {
            char_counts[c - 'a']++;
        }

        // Try removing one instance of each possible character
        for (int i = 0; i < 26; ++i) {
            if (char_counts[i] > 0) {
                // Simulate removal
                char_counts[i]--;

                // Check if all remaining non-zero frequencies are equal
                int target_freq = -1;
                bool possible = true;
                
                for (int j = 0; j < 26; ++j) {
                    if (char_counts[j] > 0) {
                        if (target_freq == -1) {
                            target_freq = char_counts[j];
                        } else if (char_counts[j] != target_freq) {
                            possible = false;
                            break;
                        }
                    }
                }

                if (possible) {
                    return true;
                }

                // Restore count for next iteration
                char_counts[i]++;
            }
        }

        return false;
    }
};

File: generated_code/python/solution_2532_2.py
from collections import Counter
from typing import List, Optional

class Solution:
    def equalFrequency(self, word: str) -> bool:
        # Count the frequency of each character in the word
        char_counts = Counter(word)
        
        # Try removing one occurrence of each unique character
        for char in char_counts:
            # Decrement the frequency of the current character
            char_counts[char] -= 1
            
            # If the character's count drops to 0, it's completely removed from the set
            if char_counts[char] == 0:
                del char_counts[char]
            
            # Check if all remaining characters have the exact same frequency
            # Using a set of the values gives the unique frequencies present
            if len(set(char_counts.values())) == 1:
                return True
            
            # Backtrack: restore the character's frequency for the next iteration
            char_counts[char] += 1
            
        return False